File: Agente Yaiwes principal/_work/backend-watchdog-full-repair-02/src/iii-sandbox/packages/sdk-python/src/iii_sandbox/port.py
from __future__ import annotations

from .client import HttpClient
from .types import PortMapping


class PortManager:
    def __init__(self, client: HttpClient, sandbox_id: str):
        self._client = client
        self._sandbox_id = sandbox_id

    async def expose(
        self,
        container_port: int,
        host_port: int | None = None,
        protocol: str | None = None,
    ) -> PortMapping:
        body: dict = {"containerPort": container_port}
        if host_port is not None:
            body["hostPort"] = host_port
        if protocol is not None:
            body["protocol"] = protocol
        data = await self._client.post(
            f"/sandbox/sandboxes/{self._sandbox_id}/ports", body
        )
        return PortMapping(**data)

    async def list(self) -> dict:
        return await self._client.get(
            f"/sandbox/sandboxes/{self._sandbox_id}/ports"
        )

    async def unexpose(self, container_port: int) -> dict:
        return await self._client.delete(
            f"/sandbox/sandboxes/{self._sandbox_id}/ports?containerPort={container_port}"
        )

File: Agente Yaiwes principal/_work/backend-watchdog-full-repair-02/src/iii-sandbox/packages/sdk-rust/src/port.rs
use std::sync::Arc;

use crate::client::HttpClient;
use crate::error::Result;
use crate::types::{PortListResponse, PortMapping, PortRemoveResponse};

pub struct PortManager {
    client: Arc<HttpClient>,
    sandbox_id: String,
}

impl PortManager {
    pub fn new(client: Arc<HttpClient>, sandbox_id: String) -> Self {
        Self { client, sandbox_id }
    }

    pub async fn expose(
        &self,
        container_port: i64,
        host_port: Option<i64>,
        protocol: Option<&str>,
    ) -> Result<PortMapping> {
        self.client
            .post(
                &format!("/sandbox/sandboxes/{}/ports", self.sandbox_id),
                Some(&serde_json::json!({
                    "containerPort": container_port,
                    "hostPort": host_port,
                    "protocol": protocol,
                })),
            )
            .await
    }

    pub async fn list(&self) -> Result<PortListResponse> {
        self.client
            .get(&format!("/sandbox/sandboxes/{}/ports", self.sandbox_id))
            .await
    }

    pub async fn unexpose(&self, container_port: i64) -> Result<PortRemoveResponse> {
        self.client
            .delete(&format!(
                "/sandbox/sandboxes/{}/ports?containerPort={container_port}",
                self.sandbox_id
            ))
            .await
    }
}
